src/cluster_legacy.c

```diff
@@ -6517,3 +6566,31 @@ int clusterAllowFailoverCmd(client *c) {
 void clusterPromoteSelfToMaster(void) {
     replicationUnsetMaster();
 }
+
+void updateNodesHealth(void) {
+    dictIterator *di;
+    dictEntry *de;
+    clusterNode *node;
+    int overall_health_changed = 0;
+    di = dictGetSafeIterator(server.cluster->nodes);
+    while((de = dictNext(di)) != NULL) {
```

Comment:
```suggestion
    dictIterator di;
    dictEntry *de;
    clusterNode *node;
    int overall_health_changed = 0;
    di = dictInitSafeIterator(&di, server.cluster->nodes);
    while((de = dictNext(di)) != NULL) {
```
You can stack allocate this iterator, then we don't need the alloc + free.